cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/replication/ReplicatorBuilder.java

```diff
@@ -56,11 +58,13 @@ private URI addCookieInterceptorIfRequired(URI uri) {
                 uriPort = 80;
             } else if ("https".equals(uriProtocol)) {
                 uriPort = 443;
-            } else {
-                throw new RuntimeException("Unknown protocol: "+uriProtocol);
             }
         }
 
+        Misc.checkArgument(uriPort >= 0,
+                String.format(Locale.ENGLISH,
+                        "Could not determine default port for protocol \"%s\"", uriProtocol));
```

Comment:
It seems a bit odd that the IllegalArgumentException would be thrown when a port couldn't be determined. Unsupported protocol message would make more sense.